### DISCUSSION TITLE
Are range requests supported for retrieving bulk data and/or instances?

### DISCUSSION THREAD

**Discussion by psavery** [dcm4che/dcm4chee-arc-light#4322] [General]
The DICOMweb standard mentions support for range requests for bulk data in [Section 6.5.5.2.1](https://dicom.nema.org/dicom/2013/output/chtml/part18/sect_6.5.html). Specifically, it says:

```
If the Range header is specified in the request, the server shall return only the specified bytes of the bulk data object. 
```

However, if I make a request for bulk data with a range header, it doesn't appear to have any effect. For instance:

```bash
curl --header "Range: bytes=100-107" http://localhost:8008/dcm4chee-arc/aets/DCM4CHEE/rs/studies/2.25.5098554257142885723564998800713257806/series/1.3.6.1.4.1.5962.99.1.853235442.409644740.1680185480946.4.0/instances/1.3.6.1.4.1.5962.99.1.853235442.409644740.1680185480946.9.0/bulkdata/00480105/0/00282000 > out
```

That seems to return the whole object, rather than just bytes 100-107.

We would particularly find it useful to support Range headers for retrieving whole DICOM instances, although that doesn't appear to be mentioned in the standard (it only seems to mention Range headers for retrieving bulk data).

**Comment by gunterze** [comment]
No, support of HTTP range requests by DICOM Web Retrieve Services is **not** implemented.
